### PR TITLE
Fix blended terms take 2

### LINKED ISSUE #15860: multi_match query gives java.lang.IllegalArgumentException: Illegal shift value, must be 0..63; got shift=2147483647
Hi team,

I get an error using a `multi_match` query with `cross_fields` type and a numeric query.
I'm using v2.1.1 on OSX, installed via Homebrew.

**Index basic data**
```
curl -XPUT http://localhost:9200/blog/post/1?pretty=1 -d '{"foo":123, "bar":"xyzzy", "baz":456}'
```

**Use a `multi_match` query with `cross_fields` type and a numeric query**
```
curl -XGET http://localhost:9200/blog/post/_search?pretty=1 -d '{"query": {"multi_match": {"type": "cross_fields", "query": "100", "lenient": true, "fields": ["foo", "bar", "baz"]}}}'
```

**Error**
```
{
  "error" : {
    "root_cause" : [ {
      "type" : "illegal_argument_exception",
      "reason" : "Illegal shift value, must be 0..63; got shift=2147483647"
    } ],
    "type" : "search_phase_execution_exception",
    "reason" : "all shards failed",
    "phase" : "query",
    "grouped" : true,
    "failed_shards" : [ {
      "shard" : 0,
      "index" : "blog",
      "node" : "0TxGVVWsSu2qX63hZdOv2w",
      "reason" : {
        "type" : "illegal_argument_exception",
        "reason" : "Illegal shift value, must be 0..63; got shift=2147483647"
      }
    } ]
  },
  "status" : 400
}
```

**Note that the error does not appear if I specify only 1 numeric field in search.**

**Stack trace**
```
Caused by: java.lang.IllegalArgumentException: Illegal shift value, must be 0..63; got shift=2147483647
	at org.apache.lucene.util.NumericUtils.longToPrefixCodedBytes(NumericUtils.java:147)
	at org.apache.lucene.util.NumericUtils.longToPrefixCoded(NumericUtils.java:121)
	at org.apache.lucene.analysis.NumericTokenStream$NumericTermAttributeImpl.getBytesRef(NumericTokenStream.java:163)
	at org.apache.lucene.analysis.NumericTokenStream$NumericTermAttributeImpl.clone(NumericTokenStream.java:217)
	at org.apache.lucene.analysis.NumericTokenStream$NumericTermAttributeImpl.clone(NumericTokenStream.java:148)
	at org.apache.lucene.util.AttributeSource$State.clone(AttributeSource.java:54)
	at org.apache.lucene.util.AttributeSource.captureState(AttributeSource.java:281)
	at org.apache.lucene.analysis.CachingTokenFilter.fillCache(CachingTokenFilter.java:96)
	at org.apache.lucene.analysis.CachingTokenFilter.incrementToken(CachingTokenFilter.java:70)
	at org.apache.lucene.util.QueryBuilder.createFieldQuery(QueryBuilder.java:223)
	at org.apache.lucene.util.QueryBuilder.createBooleanQuery(QueryBuilder.java:87)
	at org.elasticsearch.index.search.MatchQuery.parse(MatchQuery.java:178)
	at org.elasticsearch.index.search.MultiMatchQuery.parseAndApply(MultiMatchQuery.java:55)
	at org.elasticsearch.index.search.MultiMatchQuery.access$000(MultiMatchQuery.java:42)
	at org.elasticsearch.index.search.MultiMatchQuery$QueryBuilder.parseGroup(MultiMatchQuery.java:118)
	at org.elasticsearch.index.search.MultiMatchQuery$CrossFieldsQueryBuilder.buildGroupedQueries(MultiMatchQuery.java:198)
	at org.elasticsearch.index.search.MultiMatchQuery.parse(MultiMatchQuery.java:86)
	at org.elasticsearch.index.query.MultiMatchQueryParser.parse(MultiMatchQueryParser.java:163)
	at org.elasticsearch.index.query.QueryParseContext.parseInnerQuery(QueryParseContext.java:257)
	at org.elasticsearch.index.query.IndexQueryParserService.innerParse(IndexQueryParserService.java:303)
	at org.elasticsearch.index.query.IndexQueryParserService.parse(IndexQueryParserService.java:206)
	at org.elasticsearch.index.query.IndexQueryParserService.parse(IndexQueryParserService.java:201)
	at org.elasticsearch.search.query.QueryParseElement.parse(QueryParseElement.java:33)
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:831)
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:651)
	at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:617)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:368)
	at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:368)
	at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:365)
	at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:350)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
```

### PR DESCRIPTION
**Pull request by nik9000**
Closes #15860

### CODE REVIEW

**Comment by nik9000** [comment]
@s1monw I was a bit overzealous with that last one. I ended up merging it without running all the tests. Something about a weekend in between. I reverted it because it was pretty destructive. Anyway, this is the code from the last one in first commit and code to fix the fallout of the first commit in the second. Mostly it works around funkiness with setting an analyzer and using numeric or date fields.

**Comment by nik9000** [comment]
I just blasted all the tags saying backport this very far into the past after talking with some other Elastic employees. This has been like this for a long time and its a fiddly change so I'd like it to soak in integration testing. Thus I'm targeting just master and 2.x branches. So the 3.0 and 2.3 releases.

File: core/src/main/java/org/elasticsearch/index/search/MatchQuery.java [hunk 1]
```diff
@@ -352,11 +370,16 @@ public Query createCommonTermsQuery(String field, String queryText, Occur highFr
...
-                Query query = fieldType.fuzzyQuery(term.text(), fuzziness, fuzzyPrefixLength, maxExpansions, transpositions);
-                if (query instanceof FuzzyQuery) {
-                    QueryParsers.setRewriteMethod((FuzzyQuery) query, fuzzyRewriteMethod);
+                try {
+                    Query query = fieldType.fuzzyQuery(term.text(), fuzziness, fuzzyPrefixLength, maxExpansions, transpositions);
+                    if (query instanceof FuzzyQuery) {
+                        QueryParsers.setRewriteMethod((FuzzyQuery) query, fuzzyRewriteMethod);
+                    }
+                    return query;
+                } catch (Exception e) {
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/search/MatchQuery.java]: it looks a bit inconsistent to catch Exception here and RuntimeException above?

**Comment by jpountz** [comment]
LGTM. I agree this fix is to subtle to be backported to bugfix releases.

**Comment by nik9000** [comment]
And cherry picked to 2.x: 1c95b75b466b6dc152959c55e36d833c9cace8f1

**Pull request merged**